### PR TITLE
perform read_parquet in a async way

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing

### CODE REVIEW

File: modin/core/io/column_stores/parquet_dispatcher.py [hunk 1]
```diff
@@ -337,12 +337,26 @@ def build_query_compiler(cls, path, columns, index_columns, **kwargs):
         storage_options = kwargs.pop("storage_options", {}) or {}
         col_partitions, column_widths = cls.build_columns(columns)
         partition_ids = cls.call_deploy(path, col_partitions, storage_options, **kwargs)
-        index, sync_index = cls.build_index(
-            path, partition_ids, index_columns, storage_options
-        )
+        if "row_group_sizes" not in kwargs:
+            # range index case
```
Comment (YarShev) [inline modin/core/io/column_stores/parquet_dispatcher.py:341]: Should this comment be moved in else branch?

File: modin/core/io/column_stores/parquet_dispatcher.py [hunk 2]
```diff
@@ -421,6 +435,18 @@ def _read(cls, path, engine, columns, **kwargs):
             if dataset.schema.pandas_metadata
             else []
         )
+        if (
+            index_columns is not []
```
Comment (YarShev) [inline modin/core/io/column_stores/parquet_dispatcher.py:439]:
```suggestion
            index_columns
```
?

File: modin/core/io/column_stores/parquet_dispatcher.py [hunk 3]
```diff
@@ -421,6 +435,18 @@ def _read(cls, path, engine, columns, **kwargs):
...
+            and len(index_columns) == 1
+            and isinstance(index_columns[0], dict)
+            and index_columns[0]["kind"] == "range"
+            and index_columns[0]["step"] == 1
+            and len(dataset.fragments) == 1
```
Comment (YarShev) [inline modin/core/io/column_stores/parquet_dispatcher.py:444]: What is the `fragments` field?

File: modin/core/io/column_stores/parquet_dispatcher.py (hunk 1 above)
Comment (anmyachev) [reply modin/core/io/column_stores/parquet_dispatcher.py:341]: yes

File: modin/core/io/column_stores/parquet_dispatcher.py (hunk 3 above)
Comment (anmyachev) [reply modin/core/io/column_stores/parquet_dispatcher.py:444]: The fragment describes how one parquet file is divided into parts, there can be many fragments when reading from a parquet directory, but this patch does not support this case.

Comment (YarShev) [reply modin/core/io/column_stores/parquet_dispatcher.py:444]: does the `fragments` field describes files itself to be read from?

Comment (anmyachev) [reply modin/core/io/column_stores/parquet_dispatcher.py:444]: yes

Comment (YarShev) [reply modin/core/io/column_stores/parquet_dispatcher.py:444]: I see, thanks

**Pull request merged**